### ISSUE TITLE
Platform support: NixOS?

### ISSUE THREAD

**Issue by water-sucks** [Mastermindzh/tidal-hifi#95]
I'm attempting to use this on NixOS right now with `appimage-run` (I'll figure out how to write a proper derivation for it later, I don't know how to package an Electron application and resources on it online are thin), and it seems to be working except that there is no sound; the application does not show up in pavucontrol,and I'm using Pipewire with ALSA, PulseAudio, and JACK all enabled. All other applications (including Tidal in Chromium) have sound. Is there any way I can troubleshoot this?

**Comment by Mastermindzh** [comment]
I have no experience with NixOs and their .. unique.. package format 😄 

You can try the snap package or the unpackaged version if you want.

**Comment by water-sucks** [comment]
I might as well! I just don't really want to have to patch all the binaries and build it from source, seems like a pain. Might have been easier if this was a Yarn package but oh well.

How are you creating the tar.gz? I'm not sure how from looking at the package.json.

**Comment by Mastermindzh** [comment]
It is done by the electron-builder.yml.
But if you want to **only** build the tar.gz you can do this:

Copy the electron-builder.deb.yml to electron-builder.tar.yml and add the following contents:
```yml
extends: ./build/electron-builder.yml
linux:
  category: Audio
  icon: ./assets/icon.png
  target:
    - tar.gz
```

Then run:

```sh
electron-builder --publish=never -c ./build/electron-builder.tar.yml
```

Alternatively, you can comment out other targets in the regular electron-build and just use npm (or yarn) build:

```yml
linux:
  category: Audio
  target:
    # - pacman
    - tar.gz
    # - deb
    # - rpm
    # - AppImage
    # - snap
    # - freebsd
```

**Comment by water-sucks** [comment]
Bit of change and I figured out why it wasn't working! Turns out I don't have the Widevine dependency it requires, so it just hung. Found this problem inadvertently while trying to use TIDAL Web on ungoogled-chromium built without Widevine and it had the same behavior, and I put two and two together.

I'm going to improve on it right now, I haven't had time to before. Hopefully by then I'll have a derivation ready to go and I'll drop it into nixpkgs for you! It's depends on the deb package though.
